File: Common/mbf_get_then_put.m
function mbf_get_then_put(pv_name, new_value)
% retrieves the value before the set_variable and then write the new value to the
% PV.
%
% example: mbf_get_then_put(pv_name, new_value)

[root_string, ~, ~, ~] = mbf_system_config;
root_string = root_string{1};

if iscell(pv_name)
    for hse = 1:length(pv_name)
        original_value = get_variable(pv_name{hse});
        set_variable(pv_name, new_value)
        save(fullfile(root_string, 'captured_config', pv_name{hse}), 'original_value')
    end %for
else
    original_value = get_variable(pv_name);
    set_variable(pv_name, new_value)
    save(fullfile(root_string, 'captured_config', pv_name), 'original_value')
end %if

File: python_port/Common/mbf_get_then_put.py
from cothread.catools import caGet, caPut
from Common import mbfSystemConfig
import os
from scipy.io import savemat


def mbfGetThenPut(pv_name, new_value):
    """retrieves the value before the caPut and then
        write the new value to the PV.
    Example: mbf_get_then_put(pv_name, new_value)
    """

    root_string, _1, _2, _3 = mbfSystemConfig()
    root_string = root_string[0]

    for hse in range(len(pv_name)):
        original_value = caGet(pv_name[hse])
        caPut(pv_name, new_value)
        savemat(
            os.path.join((root_string, "captured_config", pv_name[hse])), original_value
        )
